src/state/session/index.tsx

```diff
@@ -618,20 +618,24 @@ export function Provider({children}: React.PropsWithChildren<{}>) {
 
       logger.debug(`session: persisted onUpdate`, {})
 
-      if (session.currentAccount && session.currentAccount.refreshJwt) {
-        if (session.currentAccount?.did !== state.currentAccount?.did) {
+      const selectedAccount = session.accounts.find(
+        a => a.did === session.currentAccount?.did,
+      )
```

Comment:
This just effectively ensures we don't read `session.currentAccount` and instead read `session.accounts.find(a => a.did === session.currentAccount?.did)`.

So `session.accounts` is always the source of truth for the actual data. However, note that `session.currentAccount?.did` plays a decisive role in whether we're going to "find" an account or not. This keeps the existing code (e.g. logout relying on setting `currentAccount` to `undefined`) working.